pkg/shell/hosts_dialog.jsx

```diff
@@ -357,7 +365,7 @@ class MachinePort extends React.Component {
                 <span>{cockpit.format(_("Unable to contact $0."), this.props.full_address)}</span>
                 <span>{_("Is sshd running on a different port?")}</span>
             </p>
-            <Form isHorizontal>
+            <Form isHorizontal onSubmit={prevent_default(callback)}>
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19018-20230627-115453-74cfaf23-fedora-38-devel-storage//Coverage/lcov/github-pr.diff.gcov.html)